spec/jobs/risc_delivery_job_spec.rb

```diff
@@ -91,25 +112,51 @@
       end
     end
 
-    context 'Errno::ECONNREFUSED error' do
+    it_behaves_like 'retries and logs event when attempts exhausted', Faraday::SSLError
+    it_behaves_like 'retries and logs event when attempts exhausted', Faraday::ConnectionFailed
+
+    context 'when the job fails due to an Errno::ECONNREFUSED error' do
       before do
+        allow_any_instance_of(described_class).to receive(:analytics).and_return(job_analytics)
         # stub_request().to_raise wraps this in Faraday::ConnectionFailed, but
         # in actual usage, the original error is unwrapped
-        expect(job.faraday).to receive(:post).and_raise(Errno::ECONNREFUSED)
+        @connection = instance_double(Faraday::Connection)
+        allow(@connection).to receive(:post).and_raise(Errno::ECONNREFUSED)
+        allow(Faraday).to receive(:new).and_return(@connection)
       end
 
-      it 'raises and retries via ActiveJob' do
-        expect { perform }.to raise_error(Errno::ECONNREFUSED)
-      end
-
-      context 'it has already failed twice' do
-        before do
-          allow(job).to receive(:executions).and_return 2
+      context 'when the job fails for the 1st time' do
```

Comment:
i don't personally really like `shared_examples`, but since you've set them up, why wouldn't you use them for the different types of errors?

Reply:
My preference has been to use `shared_examples` for larger examples.

Reply:
since they are designed for reuse, it is confusing to me that you would add them and then not use them for all the similar specs -- it makes me think these are different in some way. am i missing something about them that makes them different?

i would prefer to either write out the longer specs explicitly (my actual preference, since i find shared examples hard to read, especially when there are failures), *or* use them for all examples because of that confusion it creates. does that make sense?

Reply:
I see what you mean. The mocking fixture was different, the call count expectation was different and the risc event payload error message was different. I'll get rid of the shared_examples as they seem to add to the confusion in this case.